pkg/util/parametermap/map.go

```diff
@@ -0,0 +1,145 @@
+package parametermap
```

Comment:
also i don't understand this name, do we want to implement ordered dict/map?

Reply:
What this map shall provide is adding new values with a prio, when the key for a value already exists a new value shall not be added if it is less important than the already existing value. Adding values to the map can happen distributed over a codebase while keeping precedence of values decoupled from execution order.

Reply:
Priority map sounds better than parameter map :) as you described.

Reply:
renamed.